.github/workflows/main.yml

```
@@ -0,0 +1,14 @@
+name: build and test
+on: [push, pull_request]
```

```suggestion
on: push
```

We want to run the script once per push and we should only be opening pull requests on things that have been pushed anyway, so running on push should be sufficient.